### ISSUE TITLE
FED - Update breadcrumbs functionality and follow GDS guidelines

### ISSUE THREAD

**Issue by MichaelGrahamVS** [visitscotland/vs-component-library#183]
Our current breadcrumbs don't follow the GDS guidelines and we need to either make this the same or justify why this is different. 

The ScotGov and GDS design system both use arrows instead of pipes to separate links.

It is bad practice to truncate links which we currently do as it may confuse a user. GDS guidelines show that links do go onto multiple lines. We should think about using this pattern. 

We may also want to follow GDS and not display the current page to help us shorten breadcrumbs. 

On mobile, GDS also show the Home link and last level so if we remove showing the current page, we may want to add the Home link in.

 

https://design-system.service.gov.uk/components/breadcrumbs/

https://designsystem.gov.scot/components/breadcrumbs/ 

Find wire frame link attached!
https://xd.adobe.com/view/63cba62b-e3e5-4a6b-bf0a-3a8add63e8b8-3698/?x_product=xd-jiraserver%2F1.0

Follow-on ticket required for CMS work 
 

**Requirements / Processing / Business Rules**

**Mobile**

 use 1st and last item of breadcrumb
Separate by arrow rather than pipe

**Desktop**

Do not truncate breadcrumbs
No maximum length to be defined and can go over multiple lines
Do not display current page in breadcrumbs
Separate by arrow rather than pipe

<html>
<body>
<div class="user-content-block" style="margin: 0px; padding: 0px; overflow: auto hidden; color: rgb(23, 43, 77); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, Cantarell, &quot;Open Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><p style="margin: 10px 0px 0px; padding: 0px;"><b>Acceptance Criteria</b></p><div class="table-wrap" style="margin: 0px; padding: 0px;">

no | Summary | Given | When | Then | Comments
-- | -- | -- | -- | -- | --
1 | breadcrumb- desktop | Breadcrumb on any | a user is on any page | the current page no longer appears in the breadcrumb. **AND** All levels (up to 6) above appear as breadcrumbs **AND** Breadcrumbs are not truncated | desktop; all languages
2 | breadcrumb- mobile | Breadcrumb on any | a user is on any page | The first breadcrumb (home) **AND** the last breadcrumb (level above) are displayed | mobile; all languages
3 | breadcrumb- - separator | Breadcrumb on any | a user is on any page | The breadcrumb is separated by an arrow  (was pipe) | mobile and desktop; all languages
4 | Tabbing order & Focus state | a user is using a keyboard to access the page | a user accesses the  page | focus moves from top left through and down to last module **AND** currently selected link focus state is always clear and obvious  (as per the design) | desktop;
5 | Hover state | a user is on any page | a user  hovers over a breadcrumb | hover state is clear and obvious **AND** the underline is removed (as per the design) | desktop; all languages
6 | Active state | a user is on any page | a user clicks a breadcrumb | active state is clear and obvious | desktop and mobile; all languages

</div><p style="margin: 10px 0px 0px; padding: 0px;"> </p></div><br class="Apple-interchange-newline">
</body>
</html>

Original Ticket 
https://jira.visitscotland.com/browse/VSD-159

**Comment by annacardellini** [comment]
Yes, so having a look at this I think the bar versus arrow isn't important its about the spacing (or padding) we have assigned to each breadcrumb. So if anything it may be that we need to allow for more space between breadcrumb and pipe to account for the increase in target size requirements brought in with the WCAG 2.2 guidance. @kyle-fraser  @TessaMinshull are you be able to check the breadcrumb bar to check if it currently passes with the new WCAG 2.2 guidance on increased target size and if not can we look to create new designs that do to see how that might look?

![image](https://github.com/visitscotland/vs-component-library/assets/143170566/4296de35-1554-4f54-8988-dac292d16f44)

**Comment by MichaelGrahamVS** [comment]
@TessaMinshull can review this ticket from a Design Perspective please?

**Comment by TessaMinshull** [comment]
Hello, based on the government application on breadcrumbs we will pass as the links are bigger than the 24 px target size. It's not to do with the space between the bar and link but the interactive elements around it. The only case we would have to worry about this would be if the breadcrumb would run over multiple lines but we couldn't see a case of this happening. 


WCAG 2.2
(Make sure no other interactive elements are too close to the breadcrumbs. This is to make sure users can easily interact with the breadcrumbs. This relates to WCAG 2.2 success criterion [2.5.8 Target Size (minimum)](https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html).

https://design-system.service.gov.uk/components/breadcrumbs/)


Was there any updates on this @annacardellini 'It is bad practice to truncate links which we currently do as it may confuse a user. GDS guidelines show that links do go onto multiple lines. We should think about using this pattern.'?

**Comment by annacardellini** [comment]
hmm might need to look into that specifically @TessaMinshull. im thinking of one example off the top of my head that i'll screenshot and add in, (for Loch Lomond, trossachs, stirling & forth valley) where if we used the GDS solution it would just shorten this to first and last which could also be confusing to users. I suppose there is a wider issue with naming here, in that for this example does it need to include all those areas? or could that be split out/ renamed to still serve the purpose (that could also be applied to any others that were being truncated). 

I would say that its only truncated on the final page (ie the Loch Lomond, trossachs, stirling & forth valley page) for which there is a whopping great H1 that sits directly below it that has it in full so not sure how confusing it'd be as users would be more likely to forget the pages previous than the page currently on. Would really need to look at more examples (that are maybe a second level of 3 in a breadcrumb) and maybe run some user tests to get a better idea. Would be keen to know with the truncated version (and the proposed GDS solution) what happens with screen readers? do they still read out in full? if so thats more promising than if its also cut in the read out.

![image](https://github.com/visitscotland/vs-component-library/assets/143170566/6eba1e47-5e66-4091-9a06-1845a69cf1b7)

**Comment by MichaelGrahamVS** [comment]
Hey @annacardellini and @TessaMinshull sounds like more conversation is needed around this. Let's put it on the backburner for now given all the strategic review work that is going on. We can revisit at a later date. :-)
